core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/git_package_auth_provider.go

```diff
@@ -0,0 +1,59 @@
+package git_package_content_provider
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"os"
+	"path"
+	"strings"
+)
+
+const (
+	// filename for storing a github auth token for a user
+	githubUserTokenFilename = "token.txt"
+
+	githubTokenFilePerms = 0644
+)
+
+type GitHubPackageAuthProvider struct {
+	// Location inside APIC where GitHub auth info exists
+	githubAuthStorageDirPath string
+}
+
+func NewGitHubPackageAuthProvider(githubAuthStorageDirPath string) *GitHubPackageAuthProvider {
+	return &GitHubPackageAuthProvider{
+		githubAuthStorageDirPath: githubAuthStorageDirPath,
+	}
+}
+
+func (gitAuth *GitHubPackageAuthProvider) StoreGitHubTokenForPackage(packageId, token string) error {
+	err := os.WriteFile(path.Join(gitAuth.githubAuthStorageDirPath, getGitHubTokenFileName(packageId)), []byte(token), githubTokenFilePerms)
+	if err != nil {
+		return err
+	}
+	logrus.Infof("Successfully stored GitHub auth token for package: '%v'", packageId)
+	return nil
+}
+
+func (gitAuth *GitHubPackageAuthProvider) GetGitHubTokenForPackage(packageId string) string {
+	tokenBytes, err := os.ReadFile(path.Join(gitAuth.githubAuthStorageDirPath, getGitHubTokenFileName(packageId)))
+	if err != nil {
+		return ""
```

Comment:
opted to not return an error here to make the upstream err handling simpler but maybe I should? or at least log it?